datafusion/common/src/scalar.rs

```diff
@@ -3419,8 +3492,8 @@ mod tests {
     {
         let scalar_result = left.add_checked(&right);
 
-        let left_array = left.to_array();
-        let right_array = right.to_array();
+        let left_array = left.to_array().expect("Failed to convert to array");
```

Comment:
```suggestion
        let left_array = left.to_array().expect("Failed to convert to array in add overflow check");
```

Reply:
I believe the backtrace will provide information on the test function. For example, the following context is provided on failure
`thread 'scalar::tests::scalar_add_overflow_test' panicked at datafusion/common/src/scalar.rs:3495:9:`